packages/inquirer-common/src/metadata/metadataHelpers.ts

```diff
@@ -0,0 +1,241 @@
+import type { ConvertedMetadata, EntitySet } from '@sap-ux/vocabularies-types';
+import { convert } from '@sap-ux/annotation-converter';
+import { parse } from '@sap-ux/edmx-parser';
+import { t } from '../i18n';
+
+/**
+ * Required transformations for analytical table support.
+ * NOTE: This constant is primarily used by odata-service-inquirer but is exported
+ * here to maintain backward compatibility with external packages that import it.
+ */
+export const transformationsRequiredForAnalyticalTable = [
+    'filter',
+    'identity',
+    'orderby',
+    'skip',
+    'top',
+    'groupby',
+    'aggregate',
+    'concat'
+] as const;
+
+/**
+ * Annotation pattern for RecursiveHierarchy.
+ */
+const RECURSIVE_HIERARCHY_ANNOTATION = 'RecursiveHierarchy';
+
+/**
+ * Finds an entity set by name in the metadata.
+ *
+ * @param metadata The metadata (edmx) of the service.
+ * @param entitySetName The name of the entity set to find.
+ * @returns The entity set if found, undefined otherwise.
+ */
+export function findEntitySetByName(metadata: ConvertedMetadata, entitySetName: string): EntitySet | undefined {
+    return metadata.entitySets.find((entitySet) => entitySet.name === entitySetName);
+}
+
+/**
+ * Checks if the given entity set has aggregate transformations.
+ * Returns true if ANY transformations are present in either entity set or entity type annotations.
+ *
+ * @param entitySet The entity set to check for aggregate transformations.
+ * @returns true if the entity set has any aggregate transformations, false otherwise.
+ */
+export function hasAggregateTransformations(entitySet: EntitySet): boolean {
+    const transformations =
+        entitySet.annotations?.Aggregation?.ApplySupported?.Transformations ||
+        entitySet.entityType?.annotations?.Aggregation?.ApplySupported?.Transformations;
+
+    return !!transformations && Array.isArray(transformations) && transformations.length > 0;
```

Comment:
Is adding `!!transformations` required? The `isArray` should handle if its null? Just means extra tests to cover this?